pkg/debuggers/remote/utils.go

```diff
@@ -37,13 +37,21 @@ func GetPort(pid int) (int, error) {
 	return port, nil
 }
 
-func GetPortOfJavaProcess(pid int) (int, error) {
+func GetPortOfJavaProcess(pid int, env map[string]string) (int, error) {
 
 	args, err := utils.GetCmdArgsByPid(pid)
 	if err != nil {
 		log.WithFields(log.Fields{"pid": pid, "err": err}).Error("Can't get command line arguments")
 		return 0, err
 	}
+	s, ok := env["JAVA_TOOL_OPTIONS"]
+	if ok {
+		s = strings.Replace(s, "\x00", " ", -1)
+		ss := strings.Split(s, " ")
+		for i := range ss {
+			args = append(args, ss[i])
```

Comment:
I believe you're appending so that `checkAndParseArgument` will run on the values read from the env. Instead, let's call `checkAndParseArgument` or similar here and return if a port is found

Reply:
Wouldn't it be more useful this way. Will be calling `checkAndParseArgument`  for those args anyways if we do with pid command. We are just augmenting command so we don't need any additional handling. Anyways I am quiet open in calling `checkAndParseArgument` again here on value of env variable.

Reply:
my concern here is mostly readablity - it gets confusing if we store command line args and environment variables in the `args` parameter. One thing you could do is bundle this new logic in a helper function

Reply:
Multiple cli's has options in providing the cli args as environment variables. We need to understand one thing here, all the variables we are referencing here from environment, there is an alternate way to supply them using args directly.  For clarity we can call the map as argsFromEvnVariables.